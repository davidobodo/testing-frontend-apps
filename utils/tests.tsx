import React, { ReactElement, ReactNode } from "react";
import { render, RenderOptions } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

const Wrapper = ({ children }: { children: ReactNode }) => {
	return <div>{children}</div>;
};

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, "wrapper">) =>
	render(ui, { wrapper: Wrapper, ...options });

const setupUserEvent = (jsx: JSX.Element) => {
	return {
		user: userEvent.setup(),
		...customRender(jsx),
	};
};

// re-export everything
export * from "@testing-library/react";

export { setupUserEvent };
