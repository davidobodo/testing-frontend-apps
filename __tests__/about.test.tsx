import About from "../pages/about";
import { render } from "../utils/tests";
describe("About page", () => {
	it("Should render with correct details.", () => {
		const { asFragment } = render(<About />);
		expect(asFragment()).toMatchSnapshot();
	});
});
