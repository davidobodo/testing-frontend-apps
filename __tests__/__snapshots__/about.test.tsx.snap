// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About page Should render with correct details 1`] = `
<DocumentFragment>
  <main
    class="main"
  >
    <h1>
      About
    </h1>
    <p>
      This page is a static page whose content doesn't change often, which makes it perfect for snapshot test
    </p>
    <p>
      Always use snapshot tests for static pages, cause that way you can know when details in the page change
    </p>
    <a
      href="/"
    >
      Go Home
    </a>
  </main>
</DocumentFragment>
`;
